bundles/org.openhab.core.addon.marketplace/src/main/java/org/openhab/core/addon/marketplace/internal/community/HtmlUnescapeUtil.java

```diff
@@ -0,0 +1,226 @@
+/**
+ * Copyright (c) 2010-2021 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.core.addon.marketplace.internal.community;
+
+import java.io.StringWriter;
+import java.util.HashMap;
+
+/**
+ * Utility class to escape the common HTML entities in the inline code blocks retrieved from Discourse
+ *
+ * {@linkplain https://stackoverflow.com/a/24575417}
+ *
+ * @author Nick Frolov - initial contribution
```

Comment:
Pfew, I doubt that he agreed to contribute it here as EPL code - if so, he needs to be mentioned as an "Also-by" in the commit. If not, I'm afraid we'll have to find some other solution...

Reply:
In addons unbescape is used. Maybe it also fits here?

Reply:
It's 174KB, so yes, we might want to make this a library that can be used throughout our code.

Reply:
I took this remark ("[Whoever finds this in google is welcome to use it.](https://stackoverflow.com/questions/994331/how-to-unescape-html-character-entities-in-java/24575417#24575417)") as the author basically putting the code into the public domain, but I realize I was wrong - the license for that particular code according to https://stackoverflow.com/help/licensing would be CC BY-SA 3.0 according to the date of the post. The other ways of unescaping HTML entities would be to use a helper found in Apache Commons (commons-text) but as its use being deprecated, it won't work either.

Looking beyond this, it seems that the only entities that are escaped (from only 3 examples so maybe to be taken with a grain of salt: https://community.openhab.org/t/connectedcar-widget/125893.json, https://community.openhab.org/t/ui-widget-miflora-flowercare-plant-sensor/109563.json & https://community.openhab.org/t/keypad-widget/123589.json) are:
- `&amp;`
- `&lt;`
- `&gt;`

So maybe it would be enough to only unescape the 5 entities defined in XML: https://en.wikipedia.org/wiki/List_of_XML_and_HTML_character_entity_references#Predefined_entities_in_XML with simple `String.replace`s, and get rid of that class altogether?

Reply:
If that's indeed all, then yes, go for a simple `String.replace`!

Reply:
Done, if it appears it's not enough then we can use unbescape instead.